File: src/main/java/com/banktech/javachallenge/service/domain/submarine/SonarResponse.java
package com.banktech.javachallenge.service.domain.submarine;

import com.banktech.javachallenge.service.domain.game.SimpleResponse;

import java.util.List;

public class SonarResponse extends SimpleResponse {
    private List<Entity> entities;

    public SonarResponse() {
    }

    public SonarResponse(String message, Integer code) {
        super(message, code);
    }

    public List<Entity> getEntities() {
        return entities;
    }

    public void setEntities(List<Entity> entities) {
        this.entities = entities;
    }
}

File: src/main/java/com/banktech/javachallenge/service/domain/submarine/SonarResponse.kt
package com.banktech.javachallenge.service.domain.submarine

import com.banktech.javachallenge.service.domain.game.Coordinate

/**
 *
 */
data class SonarResponse(val entities: List<Submarine>, val message: String, val code: Int)


data class Submarine(val type: String,
                     val id: Int,
                     val position: Coordinate,
                     val owner: Owner,
                     val velocity: Int,
                     val angle: Double)


